### PR TITLE
add type safety to default props

### PR DESCRIPTION
**Pull request by jschuler**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/3535

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx [hunk 1]
```diff
@@ -41,7 +42,7 @@ export class AboutModal extends React.Component<AboutModalProps, ModalState> {
   ariaLabelledBy = `pf-about-modal-title-${this.id}`;
   ariaDescribedBy = `pf-about-modal-content-${this.id}`;
 
-  static defaultProps = {
+  static defaultProps: PickOptional<AboutModalProps> = {
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx]: Why can't we do `static defaultProps = { ... } as AboutModalProps`?

Comment (jschuler) [reply packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx]: You probably can, but if one of the types was wrong, for example if you defaulted isOpen to a string `isOpen: 'false'` the TS error/warning would not be as clear

File: packages/patternfly-4/react-core/src/components/Radio/Radio.tsx [hunk 2]
```diff
@@ -32,11 +32,11 @@ export interface RadioProps
 }
 
 export class Radio extends React.Component<RadioProps> {
-  static defaultProps = {
+  static defaultProps: PickOptional<RadioProps> = {
     className: '',
     isDisabled: false,
     isValid: true,
-    onChange: Function.prototype
+    onChange: () => {}
```
[the remark this answers is not in the document]
Comment (jschuler) [reply packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: This is not the type definition but just a default value which is a stand in for any function, e.g.
```
(something) => {
  // function body
}
```

The actual type is `onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;`

[the remark this answers is not in the document]
Comment (dlabrecq) [reply packages/patternfly-4/react-core/src/components/Radio/Radio.tsx]: Sorry, missed that

**Pull request merged**